pkg/drift/terraform/terraform.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package terraform
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/abcxyz/guardian/pkg/drift/assets"
+	"github.com/abcxyz/guardian/pkg/drift/gcs"
+	"github.com/abcxyz/guardian/pkg/drift/iam"
+)
+
+const (
+	UnknownParentID   = "UNKNOWN_PARENT_ID"
+	UnknownParentType = "UNKNOWN_PARENT_TYPE"
+)
+
+// ResourceInstances represents the JSON terraform state IAM instance.
+type ResourceInstance struct {
+	Attributes struct {
+		ID      string   `json:"id"`
+		Members []string `json:"members,omitempty"`
+		Member  string   `json:"member,omitempty"`
+		Folder  string   `json:"folder,omitempty"`
+		Project string   `json:"project,omitempty"`
+		Role    string   `json:"role,omitempty"`
+	}
+}
+
+// TerraformState represents the JSON terraform state.
+type TerraformState struct {
+	Resources []struct {
+		Type      string             `json:"type"`
+		Instances []ResourceInstance `json:"instances"`
+	} `json:"resources"`
+}
+
+type Parser struct {
+	gcs             *gcs.Client
+	gcpAssetsByID   map[string]assets.HierarchyNode
```

Comment:
If you don't end up taking the interface approach, all instances of the HierarchyNode struct should be pointers.

Reply:
Done